packages/edit-site/src/components/site-editor-routes/styles-stylebook.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import Editor from '../editor';
+import { unlock } from '../../lock-unlock';
+import SidebarNavigationScreenGlobalStyles from '../sidebar-navigation-screen-global-styles';
+import GlobalStylesUIWrapper from '../sidebar-global-styles-wrapper';
+import { StyleBookPreview } from '../style-book';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+function MobileGlobalStylesUI() {
+	const { query = {} } = useLocation();
+	const { canvas } = query;
+
+	if ( canvas === 'edit' ) {
+		return <Editor />;
+	}
+
+	return <GlobalStylesUIWrapper />;
+}
+
+export const stylesStylebookRoute = {
+	name: 'styles-stylebook',
+	path: '/styles-stylebook',
```

Comment:
Do we need a second stylebook route?

What I'm wondering is if `/styles` could work with sub-routes to make it more flexible later if further subroutes are added (I'm thinking revisions). 

The route would end up looking like:

`?p=/styles/style-book`

```js
	areas: {
		content( { query, params } ) {
			console.log( { query, params } );
			// params.subRoute, e.g., `style-book`
			// query.section, e.g., `/typography`
                        // Now handle the logic in the component as per usual
			return (
				<GlobalStylesUIWrapper
					subRoute={ params.subRoute }
					section={ query.section }
				/>
			);
		},
```

Reply:
Revisions is also meant to work with the style book though? How would that look if they're both sub-routes of styles? Could revisions instead be a sub-route of _either_ styles or styles-stylebook?

I don't feel strongly about either option. Mostly at this point I'm thinking about the further splitting up of stylebook into a few sub-routes equivalent to the tabs in the older version. But that should still work if stylebook is itself a sub-route so 🤷

Reply:
I think we need to think about this more holistically. The way I think about routes is that I try to map these to navigation menus:

 - The "root" sidebar of the site editor should represent the top level routes `/some-top-level-section`
 - If there's a sub-menu within one of these section, I should create a sub route `/top-level/second-level`
 - If there's multiple "modifiers" that are not necessarily a navigation menu (like switching the "mode" or "layout" or things like that) and we want this state to be url accessible, I should be using "query") `/top-level/second-level?modifier=something`

---

Based on this, you should be able to make the right call. I'm not wrong, today the "style book" is part of a dropdown where we switch what to "visualize" when editing styles. So it seems close to me to a "modifier" than a "submenu", so I'd probably use query: `/styles?preview=style-book` or something.

Reply:
Ok trying to think this through. 

What we have right now is a top level route at the root of the site editor: `/site-editor.php` and then we're using queries for everything else, e.g. Pages: `site-editor.php?p=%2Fpage`, Styles: `site-editor.php?p=%2Fstyles`. 

We _kind of_ have submenus already in that within Pages we have Published, Drafts etc. I guess within Styles we could consider the main global styles sections: Colors, Typography etc. to also be submenus. But we're not using `/top-level/second-level` URL structure for those, we're just using query parameters.

Do we plan to change the overall URL structure for these pages anytime soon? Otherwise, keeping to the logic of the current structure, both submenus and modifiers should be expressed as query parameters.

Does that make sense, or am I missing anything here? I'm not really invested in any solution, just trying to understand what will work best overall!

Reply:
> What we have right now is a top level route at the root of the site editor: /site-editor.php and then we're using queries for everything else, e.g. Pages: site-editor.php?p=%2Fpage, Styles: site-editor.php?p=%2Fstyles.

I wouldn't consider the `site-editor.php` as top level route, It's just a temporary technical limitation while we iterate on admin redesign. In fact we already explored pretty permalinks in the admin and we know it's possible.

Instead consider `p` as the "path" and the rest as `query`. This is exactly how the router works btw.

Reply:
> The way I think about routes is that I try to map these to navigation menu
> Instead consider p as the "path" and the rest as query. This is exactly how the router works btw.

Thanks for sharing that. Make sense to me. 


> So it seems close to me to a "modifier" than a "submenu", so I'd probably use query: /styles?preview=style-book or something.

➕

Reply:
Ok I've updated to use a query instead.